### DISCUSSION TITLE
Fix `colour.scheme()` issues using `input.colours` in `create.heatmap`

### DISCUSSION THREAD

**Discussion by maotian06** [uclahs-cds/public-R-BoutrosLab-plotting-general#94] [Ideas]
Here I want to show how to fix one `colour.scheme` issue in `create.heatmap()` function. Credit to @stefaneng  for his smart solution:
The problem was about a duplicated color when I was trying to plot a collection of heatmaps using 10 different colors. With the old code:
```
isubgen.colours <- c(default.colours(num.clusters,'survival')[1:2],'gold3',default.colours(num.clusters,'survival')[4:num.clusters]);
plots <- list();
plot.idx <- 1;
for(i in 1:length(unique(iSubGen.class[,paste0('num_subtypes_',num.clusters)]))) {

	plots[[plot.idx]] <- create.heatmap(
		iSubGen.class[iSubGen.class[,paste0('num_subtypes_',num.clusters)] ==  i,rep(paste0('num_subtypes_',num.clusters),2)],
		same.as.matrix = TRUE,
		colour.scheme = isubgen.colours,
		at = seq(0.5,num.clusters+0.5),
		cluster.dimensions = 'none',
		print.colour.key = FALSE,
		xaxis.tck = 0,
		yaxis.tck = 0,
		colourkey.cex = 0.01,
		axes.lwd = 1,
		top.padding = 0,
		bottom.padding = 0,
		axis.xlab.padding = 0,
		covariates.top.padding = 0,
		right.padding = 0,
		left.padding = 0
		);
	plot.idx <- plot.idx + 1;
}
create.multipanelplot(
	plots,
        layout.width = 1,
	file = 'test.png'
);
```
It always shows a duplicated color at the end and the color doesn't match what the `colour.scheme` allocated:
<img width="40" alt="image" src="https://user-images.githubusercontent.com/47042983/218600692-010b4de6-f4ed-49d3-8de3-186d0d1c2679.png">

@stefaneng 's solution was using the `input.colours = TRUE` option:
```
plots[[plot.idx]] <- create.heatmap(
		data.frame(rep(isubgen.colours[i],length(which(iSubGen.class[, paste0('num_subtypes_',num.clusters)] ==  i))),
					rep(isubgen.colours[i],length(which(iSubGen.class[, paste0('num_subtypes_',num.clusters)] ==  i)))),
		same.as.matrix = TRUE,
		input.colours = TRUE,
		clustering.method = 'none',
		print.colour.key = FALSE,
		xaxis.tck = 0,
		yaxis.tck = 0,
		colourkey.cex = 0.01,
		axes.lwd = 1,
		top.padding = 0,
		bottom.padding = 0,
		axis.xlab.padding = 0,
		covariates.top.padding = 0,
		right.padding = 0,
		left.padding = 0
		);
```
Problem resolved:
<img width="40" alt="image" src="https://user-images.githubusercontent.com/47042983/218601242-681da933-ef32-4485-9ebd-fee026594468.png">

Not sure why.....

**Comment by stefaneng** [comment]
@maotian06 I like using `input.colours = TRUE` more but I think your first example is not working because you do no have `total.colours = length(isubgen.colours) + 1`? Not sure if this will fix it but I always have issues without setting `total.colours`
